components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/APIDefinitionFromSwagger20.java

```diff
@@ -0,0 +1,331 @@
+            } else {
+                resource = registry.get(resourcePath);
+            }
+            resource.setContent(apiDefinitionJSON);
+            resource.setMediaType("application/json");
+            registry.put(resourcePath, resource);
+
+            //@todo Set visibility as same as the API visibility
+            //Need to set anonymous if the visibility is public
+            APIUtil.setResourcePermissions(apiProviderName, null, null, resourcePath);
+
+        } catch (RegistryException e) {
+            handleException("Error while adding Swagger Definition for " + apiName + "-" + apiVersion, e);
+        } catch (APIManagementException e) {
+            handleException("Error while adding Swagger Definition for " + apiName + "-" + apiVersion, e);
+        }
+
+    }
+
+
+    /**
+     * This method returns api definition json for given api
+     *
+     * @param apiIdentifier api identifier
+     * @param registry      user registry
+     * @return api definition json as json string
+     * @throws APIManagementException
+     */
+    @Override
+    public String getAPIDefinition(APIIdentifier apiIdentifier, Registry registry) throws APIManagementException {
+        String resourcePath = APIUtil.getSwagger20DefinitionFilePath(apiIdentifier.getApiName(),
+                apiIdentifier.getVersion(), apiIdentifier.getProviderName());
+
+        JSONParser parser = new JSONParser();
+        JSONObject apiJSON;
+        String apiDefinition = null;
+        try {
+            if (registry.resourceExists(resourcePath + SWAGGER_2_0_FILE_NAME)) {
+                Resource apiDocResource = registry.get(resourcePath + SWAGGER_2_0_FILE_NAME);
+                String apiDocContent = new String((byte[]) apiDocResource.getContent());
+                apiJSON = (JSONObject) parser.parse(apiDocContent);
+                apiDefinition = apiJSON.toJSONString();
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("Resource " + SWAGGER_2_0_FILE_NAME + " not found at " + resourcePath);
+                }
+            }
+        } catch (RegistryException e) {
+            handleException("Error while retrieving Swagger v2.0 Definition for " + apiIdentifier.getApiName() + "-" +
+                    apiIdentifier.getVersion(), e);
+        } catch (ParseException e) {
+            handleException("Error while parsing Swagger v2.0 Definition for " + apiIdentifier.getApiName() + "-" +
+                    apiIdentifier.getVersion() + " in " + resourcePath, e);
+        }
+        return apiDefinition;
+    }
+
+    /**
+     * This method generates swagger 2.0 definition to the given api
+     *
+     * @param api api
+     * @return swagger v2.0 doc as string
+     * @throws APIManagementException
+     */
+    @Override
+    @SuppressWarnings("unchecked")
+    public String createAPIDefinition(API api) throws APIManagementException {
+        JSONParser parser = new JSONParser();
```

Comment:
This is an overridden method and the method name changed to generateAPIDefinition()
There are two implementations for APIDefinition[1] as APIDefinitionFromSwagger12 and APIDefinitionFromSwagger20.

The code has been changed in [2]

[1] - https://github.com/madusankapremaratne/carbon-apimgt/blob/release-1.9.1/components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIDefinition.java
[2] - https://github.com/wso2/carbon-apimgt/pull/217/